### ISSUE TITLE
UseShellExecute doesn't throw if fails on Linux/macOS

### ISSUE THREAD

**Issue by SteveL-MSFT** [dotnet/runtime#49788]
### Description

When using `Process.Start()` where `ProcessStartInfo.UseShellExecute = true` on Unix systems, the dotnet runtime relies on xdg-open (etc...) on Linux and open on macOS to call the shell to start that process.  However, if those commands fail, you get stderr written directly to the console.  Instead the exit code should be checked and if fails, then an exception containing stderr should be thrown.

```csharp
    class Program
    {
        static void Main(string[] args)
        {
            var p = new Process();
            var psi = new ProcessStartInfo();
            psi.FileName = "foo";
            psi.UseShellExecute = true;
            p.StartInfo = psi;
            p.Start();
        }
    }
```

In this example, it expects `foo` to not exist.  Instead of an exception, you get this written directly to the console:

```output
The file /Users/steve/test/useshellexecute/foo does not exist.
```

If you run `open foo` directly on macOS, you see the same error message written to stderr and the exit code is `1`.

### Configuration

6.0.100-preview.2.21155.3

### Regression?

No

**Comment by danmoseley** [comment]
I think in general we do not know whether "foo" is a file in this case. It can be a URL, for example. How can we distinguish this case, from simply launching a process that returned 1?

**Comment by iSazonov** [comment]
> It can be a URL, for example.

But it is true for Windows too.

**Comment by mklement0** [comment]
I think we should throw an _exception_ in response to a nonzero exit code from `open` (macOS) / `xdg-open` (Linux), which would be the equivalent of the ShellExecute WinAPI function reporting failure (which apparently _does_ already get reported as an exception).

A nonzero exit code from these utilities signals _fundamental_ failure to interpret the operand, as opposed to whether the ultimate operation fails _once successfully delegated to the GUI shell_, which is not `open` / `xdg-open`'s responsibility:

In practice, this means that `xdg-open` and `open` fail (report a nonzero exit code) ONLY:

* If a given operand starts with [what looks like] an _unsupported_ URI protocol specifier, e.g. `xdg-open nosuch://bogus`
  * As an inconsequential aside: 
    * `xdg-open` does interpret this as a _URI_ and reports non-support for this protocol with `The specified location is not supported`
    * `open` falls back to interpreting this as a _file_.
  * By contrast, Windows takes a more hopeful approach and, instead of failing, offers to look in the Microsoft Store for applications that can handle such URIs - from the perspective of ShellExecute this is _not_ a failure, so no exception occurs.

* In the absence of a what looks like a protocol specifier, the operand is assumed to be _file_ or _folder_, and both utilities fail if no such file-system item exists (e.g., `xdg-open nosuchfile`); ditto on Windows.

In short: A failure reported by whatever underlying system "function" is used - which on Unix, means a nonzero exit code from an external utility - should translate into an exception - as it already does on Windows.

**Comment by danmoseley** [comment]
@mklement0 that makes sense. I'll let the crew for this area make a determination.

Looks like Mono has the same behavior that we currently do: https://github.com/mono/mono/blob/79b6e3f256a59ede74596ce82547f320bf1e9a99/mcs/class/System/Test/System.Diagnostics/ProcessTest.cs#L365-L368
